### ISSUE TITLE
Флиппер и твердотельный гироскоп

### ISSUE THREAD

**Issue by Avtopolet** [flipperdevices/flipper-hackathon-moscow#19]
### Вы капитан команды?

- [X] Yes, I am the leader of my team

### Вы готовы приехать на Хакатон в Москву?

- [X] Yes, my team will attend

### Подробно опишите идею вашего проекта

Хочу подключить к  Флипперу гироскоп типа MPU6050, написать приложение которое будет получать сырые данные с гироскопа и преобразоввывать их в углы(yaw, pitch, roll). 

### Готовый проект будет уметь...

Оценивая свои силы реально, для меня, критерием успеха будет вывести значения углов ориентации Флиппера в пространстве в градусах на его милый экранчик. Вторая цель-передача этих углов на другое устройство (ноутбук) и управление ориентацией объекта в ..скажем Blender.  Вращая Флиппером вращать модель. 

### GitHub второго участника
https://github.com/okalachev

**Comment by factoryhero** [comment]
Это интересная идея. Если на внешний модуль добавить пару кнопок, то Флиппер можно превратить в Air Mouse или даже в 3DoF геймпад, как в Nintendo Switch.

**Comment by alexklgn** [comment]
В прошивке есть зачатки BLE HID, было бы очень круто реализовать Air Mouse. Но модуль с IMU нужно подготовить заранее — на месте не будет оборудования.

**Comment by Avtopolet** [comment]
Да, конечно, платку с модулем подготовлю.

**Comment by anna-oake** [comment]
Подготовьте, пожалуйста, принципиальную схему для модуля, присоединяемого к гребёнке. Тогда одобрим проект и заранее пришлём вам протоборды для удобного прототипирования, чтобы выглядело хорошо и вставлялось удобно.

**Comment by Avtopolet** [comment]
Модуль подключается напрямую, ток потребления модуля 500 мкА, все емкости на модуле не более одной мкФ. Замеры между VCC и GND показали не более 200 пФ, так что индуктивность не нужна.
![3](https://user-images.githubusercontent.com/94665556/142687038-9ec3e114-4927-4c74-b99f-85fa2785f542.png)
![1](https://user-images.githubusercontent.com/94665556/142687033-3ad35639-8df9-4d10-a86a-62fb47cd789e.png)
![2](https://user-images.githubusercontent.com/94665556/142687037-7cc6ca7d-5247-43e3-b24f-0551eefae129.png)

**Comment by DrZlo13** [comment]
Выглядит работоспособно, если на модуле есть подтяжки

**Comment by lomalkin** [comment]
У вас модуль питается от 3.3V выхода Флиппера, но согласно нагугленной мною схеме самого модуля гироскопа - там есть LDO, а значит, что предполагается его работа от повышенного напряжения.
![image](https://user-images.githubusercontent.com/1192479/142699403-7954e699-374d-4f5e-958f-35263e31ea81.png)
Варианты:
1. Продолжать использовать схему как у вас, но при сборке "соплей" закоротить вход и выход линейника на плате самого модуля, оставив конденсаторы и все остальное как есть. (я бы выбрал этот вариант)
2. Запитать модуль гироскопа от 5V выхода Флиппера. Логика его работы следующая: на нем всегда присутствует 5V, если питание приходит на Флиппер по USB, либо этот выход можно принудительно программно включить при работе от батареи из прошивки или графического интерфейса.
![image](https://user-images.githubusercontent.com/1192479/142699155-1f91b1a1-920f-4e8f-91a2-dd6846cdc91c.png)

**Comment by anna-oake** [comment]
❗️При разработке внешнего модуля вам могут сильно помочь эти инструкции и чертежи. Обратите внимание, там есть ссылки на исходники шаблона модуля в EasyEDA и файлы контуров плат. 

https://docs.flipperzero.one/ru/development/external-module/guide

https://docs.flipperzero.one/ru/development/ext-modules

**Comment by Avtopolet** [comment]
Именно этими шаблонами я и воспользовался

**Comment by Avtopolet** [comment]
> ❗️При разработке внешнего модуля вам могут сильно помочь эти инструкции и чертежи. Обратите внимание, там есть ссылки на исходники шаблона модуля в EasyEDA и файлы контуров плат.
> 
> https://docs.flipperzero.one/ru/development/external-module/guide
> 
> https://docs.flipperzero.one/ru/development/ext-modules

Именно этими шаблонами я и воспользовался

**Comment by Avtopolet** [comment]
> У вас модуль питается от 3.3V выхода Флиппера, но согласно нагугленной мною схеме самого модуля гироскопа - там есть LDO, а значит, что предполагается его работа от повышенного напряжения. ![image](https://user-images.githubusercontent.com/1192479/142699403-7954e699-374d-4f5e-958f-35263e31ea81.png) Варианты:
> 
> 1. Продолжать использовать схему как у вас, но при сборке "соплей" закоротить вход и выход линейника на плате самого модуля, оставив конденсаторы и все остальное как есть. (я бы выбрал этот вариант)
> 2. Запитать модуль гироскопа от 5V выхода Флиппера. Логика его работы следующая: на нем всегда присутствует 5V, если питание приходит на Флиппер по USB, либо этот выход можно принудительно программно включить при работе от батареи из прошивки или графического интерфейса.
>    ![image](https://user-images.githubusercontent.com/1192479/142699155-1f91b1a1-920f-4e8f-91a2-dd6846cdc91c.png)

Спасибо за ответ, попробывал запитать от 3.3 выхода blue pill, работает. Судя по всему стоит этот стабилизатор (https://pdf1.alldatasheet.com/datasheet-pdf/view/74628/MICREL/MIC5205-3.3BM5.html) и он будет давать 3.3 Вольта

**Comment by Avtopolet** [comment]
> ❗️При разработке внешнего модуля вам могут сильно помочь эти инструкции и чертежи. Обратите внимание, там есть ссылки на исходники шаблона модуля в EasyEDA и файлы контуров плат.
> 
> https://docs.flipperzero.one/ru/development/external-module/guide
> 
> https://docs.flipperzero.one/ru/development/ext-modules
Так, согласно документации у десятипиновой гребёнки должны быть объединены земли.
![4](https://user-images.githubusercontent.com/94665556/142720877-dec12883-5cdb-4704-abf8-86c1dc30cc3a.png)

**Comment by Avtopolet** [comment]
> Выглядит работоспособно, если на модуле есть подтяжки

Рассмотрел модуль, лини данных и тактирования подтянуты. В принципе схема совпадает с моим модулем.

**Comment by Avtopolet** [comment]
Нечаянно закрыл...

**Comment by anna-oake** [comment]
Отлично. Вам понадобятся протоборды (просто макетки по контуру модуля) или будете изготавливать сразу свои под модуль?

**Comment by Avtopolet** [comment]
>Воскресенье, 21 ноября 2021, 19:01 +03:00 от Anna Prosvetova ***@***.***>:
>
>Отлично. Вам понадобятся протоборды (просто макетки по контуру модуля) или будете изготавливать сразу свои под модуль?
>—
>You are receiving this because you modified the open/close state.
>Reply to this email directly,  view it on GitHub , or  unsubscribe .
>Triage notifications on the go with GitHub Mobile for  iOS or  Android .

Буду делать методом тёплой ламповой фрезеровки на балконе. Но протобод взял бы без использования с возвратом (приложить-посмотреть к с воей поделке), может взять(если есть в комплекте) вот эти удлиненные гребёнки.

**Comment by anna-oake** [comment]
Да, гребёнки эти трудно найти.
Подскажите, как с вами в Телеграме связаться?
Решим по поводу доставки.

**Comment by anna-oake** [comment]
UPD: Насчет гребенок ошиблась. У нас теперь гребенки стандартной длины везде используются, достать должно быть не трудно.

Но в любом случае протоборды и гребенки можем выдать, жду контакты.

**Comment by Avtopolet** [comment]
Меня можно найти по имени пользователя @vtopolet в Телеграмме.

**Comment by okalachev** [comment]
Имею опыт работы с IMU — занимаюсь квадрокоптерами. Делал estimation ориентации с MPU-9250.

Возможно, смог бы помочь.

**Comment by anna-oake** [comment]
Мы обновили 3D модели и чертежи Флиппера, теперь они соответствуют продакшн-версии, которая будет у людей на руках. На эти модели и чертежи можно ориентироваться при создании внешних модулей.

https://github.com/flipperdevices/flipperzero-3d-models

**Comment by Avtopolet** [comment]
Прошу включить okalachev в команду, если это возможно.

**Comment by anna-oake** [comment]
@Avtopolet, информацию приняли, спасибо. Отредактируйте описание ишью, там можно внести ник человека.

**Comment by anna-oake** [comment]
### Ждём ваше финальное решение

**Если вы не идёте на Хакатон**, напишите комментарий здесь и закройте issue.

**Если изменилось число участников**, напишите здесь, кого нужно исключить.

**Если вы ТОЧНО идёте на Хакатон**, каждый участник команды должен:
1. Зарегистрироваться на мероприятие [по ссылке](https://neuronspace.timepad.ru/event/1860939/)
2. Сообщить здесь, в issue, о прохождении регистрации

**Comment by okalachev** [comment]
Зарегистрировался.

**Comment by Avtopolet** [comment]
Мы идём, зарегистрированы.

**Comment by Avtopolet** [comment]
У нас там нет случайно проблемки? Ато "билеты не выданы".

**Comment by Avtopolet** [comment]
У нас там нет случайно проблемки? Ато "билеты не выданы".

**Comment by anna-oake** [comment]
Все хорошо, мне просто нужно добраться до компа. Олегу уже выдала :) Ждите в течение часа
